src/os/bluestore/bluefs_types.h

```diff
@@ -111,11 +117,11 @@ struct bluefs_fnode_t {
   template<typename T, typename P>
   friend std::enable_if_t<std::is_same_v<bluefs_fnode_t, std::remove_const_t<T>>>
   _denc_friend(T& v, P& p) {
-    DENC_START(1, 1, p);
+    DENC_START(2, 2, p);
```

Comment:
I think this should depend on `bluefs_fnode_t.type`.
legacy => 1,1,
wal_v2 => 2,2, 
Otherwise draining of WAL (for downgrade) will never be possible.

Reply:
good catch